src/tests/utests/xi_utest_data_desc.c

```diff
@@ -244,6 +244,259 @@ XI_TT_TESTCASE_WITH_SETUP(
         tt_fail();
     } )
 
+#ifdef XI_MEMORY_LIMITER_ENABLED
+XI_TT_TESTCASE(
+    utest__xi_make_desc_from_string_copy__valid_data__not_enough_memory_for_buffer,
+    {
+        /* size values */
+        size_t num_characters     = 700;
+        xi_state_t ret_state      = XI_STATE_OK;
+        char* origin_string       = NULL;
+        xi_data_desc_t* data_desc = NULL;
+        
+        const size_t allocation_space   = num_characters * sizeof( char ) +
+                                          sizeof( xi_data_desc_t ) +
+                                          sizeof( xi_data_desc_t ) +
+                                          2*sizeof( xi_memory_limiter_entry_t );
+        const size_t allocation_footprint = num_characters* sizeof( char ) +
+                                            sizeof( xi_memory_limiter_entry_t );
+        const size_t memory_limit_size = XI_MEMORY_LIMITER_SYSTEM_MEMORY_LIMIT +
+                                            allocation_space;
+
+        /* assumptions */
+        tt_assert( XI_MEMORY_LIMITER_SYSTEM_MEMORY_LIMIT > allocation_footprint );
```

Comment:
what does this assert mean? System memory limit must exceed allocation footprint.
System memory limit is the minimal amount of memory the libxively needs for minimal functionality.
Allocation footprint is a memory limiter entry + the test string size.
How does this two come together with this relation?

Reply:
I can add a comment to what it's intention is.